### ISSUE TITLE
`adjacent-overload-signatures` false positive with symbol names

### ISSUE THREAD

**Issue by janslow** [palantir/tslint#1831]
### Bug Report

- __TSLint version__: `4.0.2`
- __TypeScript version__: `2.0.10`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
class Foo {
  readonly [Symbol.toStringTag] = 'Foo';

  bar() {
    throw new Error();
  }

  [Symbol.iterator]() {
    throw new Error();
  }
}

```

with `tslint.json` configuration:

```json
{
  "rules": {
    "adjacent-overload-signatures": true
  }
}
```

#### Actual behavior

The adjacent-overload-signatures rule appears to coerce method names which are symbols (e.g., `Symbol.iterator`) into the `"undefined"`, so it can't thinks they are overloads instead of separate methods.

```
src/test.ts[8, 3]: All 'undefined' signatures should be adjacent
```

#### Expected behavior

The rule should not raise an error, as they are separate methods, not overloads.
